### DISCUSSION TITLE
Resources amounts in the GUI are not updated when you buy stuff

### DISCUSSION THREAD

**Discussion by mmavova** [ihhub/fheroes2#3785] [Ideas]
1. Go to your castle.
2. Note how much gold you have.
3. Buy a building or some troops.
4. Notice that the gold amount in the bottom right corner did not change.

5. It is the same for other resources. 

The amounts are only updated when you leave the castle screen.

**Comment by ihhub** [comment]
Hi @mmavova , are you sure about this? Did you wait till building/army appearance animation completion?

**Comment by oleg-derevenetz** [comment]
I don't consider this as a bug. Original HoMM2 doesn't allow you to see those numbers, and HoMM3 have the same behavior: resource numbers are updated only when you leave the castle/market/dwelling/etc windows.

**Comment by oleg-derevenetz** [comment]
@ihhub I understand this so that it's about resource amounts on adventure screen.

**Comment by Branikolog** [comment]
I think we should add this improvement in future: update resource and castle/hero state in lists in status window out of the castle screen. (If we are not going to stretch this window. ;) )

**Comment by mmavova** [comment]
> Hi @mmavova , are you sure about this? Did you wait till building/army appearance animation completion?

https://imgur.com/a/j94zOha

**Comment by mmavova** [comment]
> I don't consider this as a bug. Original HoMM2 doesn't allow you to see those numbers, and HoMM3 have the same behavior: resource numbers are updated only when you leave the castle/market/dwelling/etc windows.

A bug in the original game series is not a reason to perpetuate it. I see no good reason to show incorrect numbers. It interferes with gameplay too, since you cannot make decisions based on the displayed amounts.

**Comment by oleg-derevenetz** [comment]
> A bug in the original game series is not a reason to perpetuate it. I see no good reason to show incorrect numbers. It interferes with gameplay too, since you cannot make decisions based on the displayed amounts.

Everywhere you need to make decisions, those resource amounts are duplicated, because in original game adventure screen is not visible at all in such situations. I agree with @Branikolog: this may be considered as a suggestion to possible future improvement, but not as a bug.

**Comment by Branikolog** [comment]
We can also try to blur or use some sort of dimming of the whole screen around castle window not to confuse players by the surroundings... 
Seems like here can happen a new discussion...
P.S. pretty sure, we already have some similar thread, but I couldn't find it in our long list...

**Comment by ihhub** [comment]
Converted the original issue into discussion.
